### PR TITLE
0.2.0: remove data class

### PR DESCRIPTION
**Pull request by zhengp0**
This is a big change, we removed the `Data` class and let the `Model` class to directly work with `DataFrame`. The `composite_models` sub-package is probably not working right now, we will plan to fix it once we finish the overhaul on the other basic components.

### CODE REVIEW

File: src/regmod/composite_models/base.py [hunk 1]
```diff
@@ -127,22 +131,29 @@ def add_offset(self, df: DataFrame, copy: bool = False) -> DataFrame:
...
+    def get_data(self) -> Optional[DataFrame]:
+        if self.df is not None:
+            return self.df.copy()
+        return None
 
     def set_data(self, df: DataFrame):
         if df.shape[0] == 0:
             raise ValueError("Attempt to use empty dataframe.")
-        self.data.attach_df(self.add_offset(df, copy=True))
+        self.df.attach_df(self.add_offset(df, copy=True))
```
Comment (davidshaw2018) [inline src/regmod/composite_models/base.py:145]: is `attach_df` a pandas method on dataframe?

File: src/regmod/data.py [hunk 2]
```diff
@@ -1,287 +0,0 @@
...
-        """Indicator of empty data frame.
-
-        Returns
-        -------
-        bool
-            Return `True` when `self.df` is empty.
-        """
-        return self.num_obs == 0
-
-    def check_cols(self) -> None:
```
Comment (davidshaw2018) [inline src/regmod/data.py:127]: are any of these validations needed, or are they added somewhere else?

**Review by davidshaw2018 — APPROVED** [review]
Like it! A lot more transparent without custom objects

File: src/regmod/composite_models/base.py (hunk 1 above)
Comment (zhengp0) [reply src/regmod/composite_models/base.py:145]: Thanks David! This is definitely wrong, will add a comment to this, when we work on `composite_model` we can solve this.

File: src/regmod/data.py (hunk 2 above)
Comment (zhengp0) [reply src/regmod/data.py:127]: I think they are definitely needed, will try to add them back. This is scheduled for v0.3.0

**Pull request merged**